db/DB.js

```diff
@@ -84,17 +94,15 @@ const DB = {
 
   async _verifySubscriber(emailHash) {
     await HIBP.subscribeHash(emailHash.sha1);
-    // TODO: resolve with error if HIBP fails
     const verifiedSubscriber = await knex("subscribers")
       .where("email", "=", emailHash.email)
       .update({ verified: true })
       .returning("*");
     return verifiedSubscriber;
   },
 
-  async removeSubscriber(email) {
+  async removeSubscriberByEmail(email) {
     const sha1 = getSha1(email);
-
     return await this._getSha1EntryAndDo(sha1, async aEntry => {
```

Comment:
I guess outside of the scope of this PR, but...
Why does it take 2 queries to delete a user? By proxying `this._getSha1EntryAndDo(sha1)`, we call `knex("subscribers").where("sha1", sha1)`, and then here we requery the same table by `knex("subscribers").where("id", "=", aEntry.id).del()`

Reply:
Yes, another refactor I'd like to perform is to remove the `_getSha1EntryAndDo` function completely.